### ISSUE TITLE
AWS Timestream support

### ISSUE THREAD

**Issue by ppbrown** [trickstercache/trickster#684]
AWS Timestream is not on the list of officially support timeseries databases.
Any plans?
Alternatively, any specific plans to exclude it?

**Comment by jakenichols2719** [comment]
To the best of my knowledge, the team doesn't have immediate plans to develop AWS Timestream support. If we take steps into AWS, the first stop is likely to be CloudWatch, since it sees significant use on any stack including AWS. Timestream isn't specifically excluded, but it's also not currently a priority.

**Comment by ppbrown** [comment]
fair enough.
it is a priority to Me though, so i’ll probably try to port it myself :)

Any pointers appreciated.

**Comment by jakenichols2719** [comment]
Right, sorry, I should have used more descriptive language! I can't speak for individuals on the team, but generally, we've got our sights on specific enhancements that we'd like to make, and our personal time goes towards those. We always have an eye out for contributors and contributions, and will happily help support development efforts.

Much of the platform support revolves around *interpretation* and *transformation* of requests. Trickster needs to be able to take a query for the last 24h of data, determined that it cached the first 12h of those, and then modify the query to only request the other half (for example). I'd strongly suggest reading through the implementations of a few TSDBs to see how this is accomplished in production use-cases.

**Comment by jranson** [comment]
AWS closed Timestream for LiveAnalytics to new customers on June 20, 2025. It's in maintenance mode, and AWS's official guidance is to migrate to Timestream for InfluxDB. Since that is real InfluxDB under the hood, and Trickster supports InfluxDB, Trickster's existing influxdb backend already covers it.
